src/lib/SX1280Driver/SX1280.cpp

```diff
@@ -579,21 +583,114 @@ int8_t ICACHE_RAM_ATTR SX1280Driver::GetRssiInst(SX12XX_Radio_Number_t radioNumb
+                RXdataBuffer_second[0] &= 0b11;
+                // if the second packet is same to the first, it's valid
+                if(memcmp(RXdataBuffer, RXdataBuffer_second, PayloadLength) == 0)
+                {
+                    isSecondRadioGotData = true;
+                }
+            }
+        }
+
+        // second radio received the same packet to the processing radio
+        gotRadio[secondRadioIdx] = isSecondRadioGotData;
+    }
+
     uint8_t status[2];
-    hal.ReadCommand(SX1280_RADIO_GET_PACKETSTATUS, status, 2, processingPacketRadio);
-    if (packet_mode == SX1280_PACKET_TYPE_FLRC) {
-        // No SNR in FLRC mode
-        LastPacketRSSI = -(int8_t)(status[1] / 2);
-        LastPacketSNRRaw = 0;
-        return;
+    uint8_t rssi[2];
+    uint8_t snr[2];
+
+    for(uint8_t i=0;i<2;i++)
+    {
+        if (gotRadio[i])
+        {
+            hal.ReadCommand(SX1280_RADIO_GET_PACKETSTATUS, status, 2, radio[i]);
+
+            if (packet_mode == SX1280_PACKET_TYPE_FLRC) 
+            {
+                // No SNR in FLRC mode
+                rssi[i] = -(int8_t)(status[1] / 2);
+                snr[i] = 0;
+            }
+            else
+            {
+                // LoRa mode has both RSSI and SNR
+                rssi[i] = -(int8_t)(status[0] / 2);
+                snr[i] = (int8_t)status[1];
+
+                // https://www.mouser.com/datasheet/2/761/DS_SX1280-1_V2.2-1511144.pdf p84
+                // need to subtract SNR from RSSI when SNR <= 0;
+                int8_t negOffset = (snr[i] < 0) ? (snr[i] / RADIO_SNR_SCALE) : 0;
+                rssi[i] += negOffset;
+            }
+
+            // If radio # is 0, update LastPacketRSSI, otherwise LastPacketRSSI2
+            (i == 0) ? LastPacketRSSI = rssi[i] : LastPacketRSSI2 = rssi[i];
+            // Update whatever SNRs we have
+            LastPacketSNRRaw = snr[i];
+        }
+    }
+
+    // by default, set the last successful packet radio to be the current processing radio (which got a successful packet)
+    instance->lastSuccessfulPacketRadio = instance->processingPacketRadio;
+
+    // when both radio got the packet, use the better RSSI one
+    if(gotRadio[0] && gotRadio[1])  
+    {
+        // LastPacketSNRRaw = (snr[0]>snr[1])? snr[0]: snr[1]; // design choice
+        LastPacketSNRRaw = (snr[0]+snr[1])/2;
+        // Update the last successful packet radio to be the one with better signal strength
+        instance->lastSuccessfulPacketRadio = (rssi[0]>rssi[1])? radio[0]: radio[1];
+    }
+
+#if defined(DEBUG_RCVR_SIGNAL_STATS)
+    // stat updates
+    for (uint8_t i = 0; i < 2; i++)
+    {
+        if (gotRadio[i])
+        {
+            instance->rxSignalStats[i].irq_count++;
+            instance->rxSignalStats[i].rssi_sum += rssi[i];
+            instance->rxSignalStats[i].snr_sum += snr[i];
+            if (snr[i] > instance->rxSignalStats[i].snr_max) { instance->rxSignalStats[i].snr_max = snr[i]; }
```

Comment:
formatting

Reply:
so you don't like in-line curly braces? :D